picarro_data/download/download_firestore.py

```diff
@@ -46,7 +47,15 @@ def main():
 
         all_days.append(pd.DataFrame.from_records(dfs))
 
-    pd.concat(all_days).to_csv(f"{args.filepath}/data_{args.date}.csv", index=False)
+    # make filepath if it doesn't exist
+    Path(args.filepath).mkdir(parents=True, exist_ok=True)
+
+    if args.end:
+        pd.concat(all_days).to_csv(
+            f"{args.filepath}/data_{args.date}_{args.end}.csv", index=False
+        )
+    else:
+        pd.concat(all_days).to_csv(f"{args.filepath}/data_{args.date}.csv", index=False)
```

Comment:
There's some unnecessary repetition here. This `if` block only seems to be used to determine the file name. So we can get away with something like this:

```suggestion
    save_path = Path(args.filepath)
    save_path.mkdir(parents=True, exist_ok=True)
    
    df_save = pd.concat(all_days)
    save_filename = f"data_{args.date}_{args.end}.csv" if args.end else f"data_{args.date}.csv"

    df_save.to_csv(save_path / save_filename, index=False)
```